client/modules/workplace/components/collection_fields/object_tree.jsx

```diff
@@ -32,17 +71,28 @@ class CollectionFields extends React.Component {
             nestLevel={nestLevel}
             updateObjectTree={this.updateObjectTree}
             updateCollectionField={updateCollectionField}
+            onClick={this.handleClick(data)}
           />
         }
         itemClassName={
           !data.nestedData ? `no-arrow nest-level-${nextLevel}` : `nest-level-${nextLevel}`
         }
-        defaultCollapsed={nestLevel > 0}
+        collapsed={data.isCollapsed}
+        onClick={this.handleClick(data)}
       >
         {data.nestedData && data.nestedData.map((item, j) => this.treeView(item, j, nextLevel))}
       </TreeView>
     );
   }
+  handleClick(field) {
+    if (field.type !== 'object') return null;
+    const { data } = this.state;
+    const self = this;
+      return () => {
+        const newData = toggleCollapse(data, field.expression);
+        self.setState({ data: newData });
+      };
+  }
   updateObjectTree() {
     this.setState({ data: this.state.data });
```

Comment:
@uapasha this case is handled there https://github.com/JSSolutions/Databazel/pull/19/files#diff-102ac9d56f00052e6853d8f544b920abR51